### ISSUE TITLE
`.o-icon-private` span missing role

### ISSUE THREAD

**Issue by aviannamiller** [omeka/omeka-s#2334]
Admin-side `.o-icon-private` icons on resources are in a span with an `aria-label` of "Private". This aria-label is not currently detected by assistive tech as the span is missing a role.

**Comment by aviannamiller** [comment]
That's great, thanks. After retesting with screen readers, I'm thinking these icons could ideally be grouped / nested with their resource names, especially since one otherwise might have quite a number of indistinguishable images labeled as "Private" amongst a given listing of resources (e.g. a user parsing for private resources within tables via the images rotor). Does this make sense / could it be done effectively?

**Comment by kimisgold** [comment]
It makes sense. Here's where I've applied that pattern:

* Resource browse
* Site browse
* Page browse within site admin context
* Site page edit, add to page list block
* Site navigation, "Add a link to a site page"

**Comment by aviannamiller** [comment]
That's great! Those private resources also now become distinguishable alongside resource names in the links rotor, so that works very well.

Icons that accompany links (like resource names) are now receiving the link styling, so notably there's a change in color there. I don't see a problem with it but just wanted to confirm that that is intended.
<img width="567" height="144" alt="Three listed item resources, the names of which are styled in red. One of the items is private, indicated by an icon of an eye with a slash through it, also styled in red." src="https://github.com/user-attachments/assets/56974b54-c662-4864-aa19-356d2b55c922" />

And, only one small issue, in site navigation ("Add a link to a site page"):
<img width="302" height="309" alt="'Add a link to a site page' sidebar section. Site titles and slugs are listed in blocks, and private sites are accompanied by private icons." src="https://github.com/user-attachments/assets/89b7963d-a54d-457e-aefa-3c726a53568d" />
The private icon aligns right against the add icon when the page title & slug overflow onto a second line. With shorter titles / slugs, the icon aligns to the right of the text. This is all appearing properly for site titles within page list blocks, which is strange considering they seem to be using similar patterns.
